code/cycler/src/wattrex_battery_cycler/mid/mid_dabs/mid_dabs.py

```diff
@@ -0,0 +1,393 @@
+class _ConstantsC:
+    PERIOD_ELECT_MEAS   = 5 # value *10ms
+    PERIOD_TEMP_MEAS    = 5 # value *10ms
+class MidDabsPwrMeterC: #pylint: disable= too-many-instance-attributes
+    '''Instanciates an object enable to measure.
+    '''
+    def __init__(self, device: list [CyclerDataDeviceC]) -> None:
+        self.device_type = [x.device_type for x in device]
+        self.dev_id: List = [x.dev_id for x in device]
+        self.bisource   : DrvEaDeviceC | None = None
+        self.source     : DrvEaDeviceC | None = None
+        self.load       : DrvRsDeviceC | None = None
+        self.epc        : DrvEpcDeviceC| None = None
+        self.meter      : DrvBkDeviceC | None = None
+        try:
+            for dev in device:
+                if dev.device_type is CyclerDataDeviceTypeE.EPC:
+                    dev_id= 0
+                    if isinstance(dev.iface_name, str):
+                        dev_id = int(dev.iface_name,16)
+                    else:
+                        dev_id = int(dev.iface_name)
+                    self.epc : DrvEpcDeviceC = DrvEpcDeviceC(dev_id=dev_id)
+                    self.epc.open()
+                    self.mapping_epc = dev.mapping_names
+                    self.epc.set_periodic(ack_en = False,
+                        elect_en = True, elect_period = _ConstantsC.PERIOD_ELECT_MEAS,
+                        temp_en = True, temp_period = _ConstantsC.PERIOD_TEMP_MEAS)
+                elif dev.device_type is CyclerDataDeviceTypeE.SOURCE:
+                    link_conf = __prepare_link_conf(dev.link_conf.__dict__)
+                    self.source : DrvEaDeviceC = DrvEaDeviceC(DrvScpiHandlerC(**link_conf))
+                    self.mapping_source = dev.mapping_names
+                elif dev.device_type is CyclerDataDeviceTypeE.LOAD:
+                    # TODO: Update SCPI not needing handler
+                    link_conf = __prepare_link_conf(dev.link_conf.__dict__)
+                    self.load : DrvRsDeviceC = DrvRsDeviceC(DrvScpiHandlerC(**link_conf))
+                    self.mapping_load = dev.mapping_names
+                elif self.device_type is CyclerDataDeviceTypeE.BISOURCE:
+                    link_conf = __prepare_link_conf(dev.link_conf.__dict__)
+                    self.bisource : DrvEaDeviceC = DrvEaDeviceC(DrvScpiHandlerC(**link_conf))
+                    self.mapping_bisource = dev.mapping_names
+                elif self.device_type is CyclerDataDeviceTypeE.METER:
+                    link_conf = __prepare_link_conf(dev.link_conf.__dict__)
+                    self.meter : DrvBkDeviceC = DrvBkDeviceC(DrvScpiHandlerC(**link_conf))
+                    self.mapping_meter = dev.mapping_names
+                else:
+                    log.error(f"The dessire device doesn't have type {self.device_type}")
+        except Exception as error:
+            log.error(error)
+            raise error
+
+    def __update_source_load_status(self, status: CyclerDataAllStatusC):
```

Comment:
Convert to a status property?

Reply:
Lets disscuss in a future version, now it´s not operational